server/src/main/java/io/druid/server/initialization/jetty/JettyServerModule.java

```diff
@@ -253,6 +255,26 @@ public void start() throws Exception
           @Override
           public void stop()
           {
+            if (activeConnections.get() != 0) {
+              long graceFulShutDownTime = (long) server.getAttribute(GRACEFUL_SHUTDOWN_TIME);
+              log.info("Waiting for [%s] milliseconds for active requests to be zero, current active requests=[%s]",
+                  graceFulShutDownTime, activeConnections.get());
+              try {
+                Thread.sleep(graceFulShutDownTime);
+              }
+              catch (InterruptedException e) {
+                log.error("Sleep has been interrupted, while waiting for active requests to be zero");
+                stopImmediately();
```

Comment:
As you added to docs, graceful shutdown is useful to wait for running queries to be completed. So, I think when an InterruptedException occurs earlier while waiting for the given ```GRACEFUL_SHUTDOWN_TIME```, we first check ```activeConnections``` is 0 and then further wait for remaining time if there remains more active connections.
What do you think?

Reply:
When an InterruptedException occurs, the thread should finish whatever it was doing as soon as possible. In this specific case , it should shutdown ASAP

Reply:
Ok. Looks good to me.